core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -480,7 +481,7 @@ public T maxTraceEvents(int maxTraceEvents) {
    *
    * @since 1.0.0
    */
-  public abstract ManagedChannel build();
+  public abstract ManagedChannel build() throws IOException;
```

Comment:
As I said before, this change is not backward compatible and thus not acceptable. Please revert, along with all the other exception handling you added elsewhere to make this possible. One of the other options is still possible, but we can't add new exceptions to pre-existing stable methods.

Reply:
This is reason I throw RuntimeException in the first place. We cannot throw at forTarget(), at build(). We cannot throw at ClientInterceptor.

It looks throwing RumtimeException is the best option, IMO.

Reply:
Why can't we throw in `forTarget()`? While the ClientInterceptor can't throw, it can fail the RPC.